benchmarking-tests/benchmark_utils.py

```diff
@@ -36,6 +44,38 @@ def run_set_statsd_host(args):
         json.dump(conf, f)
 
 
+def run_gather_metrics(args):
+    # setup a rethinkdb connection
+    conn = r.connect(args.bigchaindb_host, 28015, 'bigchain')
+
+    # setup csv writer
+    csv_file = open(args.csvfile, 'w')
+    csv_writer = csv.writer(csv_file)
+
+    # query for the number of transactions on the backlog
+    num_transactions = r.table('backlog').count().run(conn)
+    logger.info('Starting gathering metrics. {} transasctions in the backlog'.format(num_transactions))
+
+    # listen to the changefeed
+    try:
+        for change in r.table('bigchain').changes().run(conn):
+            # check only for new blocks
+            if change['old_val'] is None:
+                block_num_transactions = len(change['new_val']['block']['transactions'])
+                time_now = time.time()
+                logger.info('{} {}'.format(time_now, block_num_transactions))
```

Comment:
Not crucial here, but as a side note is better to use:
```python
logger.info('%s %s', time_now, block_num_transactions)
```
In this way you postpone the *formatting* to the logger function. Inside the logger function there is some logic to decide if showing the log or not, if the answer is yes then the string is constructed, otherwise no, saving time.

(I'll note it down and I can review the BigchainDB code later, since I'm not sure if we are doing the right thing there, and it's much more important than here)